src/ccnl-riot/src/ccn-lite-riot.c

```diff
@@ -602,11 +602,35 @@ ccnl_set_cache_strategy_remove(ccnl_cache_strategy_func func)
     _cs_remove_func = func;
 }
 
+void
+ccnl_set_cache_strategy_cache(ccnl_cache_strategy_func func)
+{
+    _cs_decision_func = func;
+}
+
 int
-cache_strategy_remove(struct ccnl_relay_s *relay, struct ccnl_content_s *c)
+cache_strategy_remove(struct ccnl_relay_s *relay, struct ccnl_content_s *c,
+                      qos_traffic_class_t *tclass)
 {
     if (_cs_remove_func) {
-        return _cs_remove_func(relay, c);
+        return _cs_remove_func(relay, c, tclass);
```

Comment:
Currently no QoS-specific logic here as most of it is already covered by allowing unsolicited content to be passed to the caching decision strategy. However, a user-defined cache replacement strategy may still require QoS information, so we pass it here.